osf/models/node.py

```diff
@@ -459,12 +464,23 @@ def is_preprint_orphan(self):
     def has_published_preprint(self):
         return self.preprints.filter(is_published=True).exists()
 
+    @property
+    def linked_preprint(self):
+        return self.preprints.get_queryset()[0]
```

Comment:
It might make sense to first try for a published preprint, then fall back to any preprint. Basically the logic in `preprint_url` (which could then just `return self.linked_preprint.url`)

Reply:
i used linked_preprint as a way to get any preprint and used it in preprint_url